base/freestanding/femr/config/keycloak/realm-data.json

```diff
@@ -16,11 +17,15 @@
     "offlineSessionIdleTimeout": 2592000,
     "offlineSessionMaxLifespanEnabled": false,
     "offlineSessionMaxLifespan": 5184000,
+    "clientSessionIdleTimeout": 0,
+    "clientSessionMaxLifespan": 0,
```

Comment:
clientSessionIdleTimeout = 0
clientSessionMaxLifespan = 0

Does this mean credentials never expire?

Reply:
Those values are now present in any newly created Keycloak realm and look to be new defaults. For that reason, I had assumed they were safe, but better safe than sorry so I looked them up: it looks like a value of `0` defers to another config.


>Client Session Idle
>If  the user is not active for longer than this timeout, refresh token  requests will bump the idle timeout. It allows for the specification of a  shorter idle timeout of refresh token than session idle timeout. And it  can be overridden on individual clients. **It is an optional configuration and if not set to a value bigger than 0 it uses the same idle timeout set in the SSO Session Idle configuration**.


>Client Session Max
>The  maximum time before a refresh token is expired and invalidated. It  allows for the specification of a shorter timeout of refresh token than  session timeout. And it can be overridden on individual clients. **It is  an optional configuration and if not set to a value bigger than 0 it  uses the same idle timeout set in the SSO Session Max configuration.**



[Server Administration Guide: Session and Token Timeouts](https://www.keycloak.org/docs/15.0/server_admin/#_timeouts)